test/rules.test.js

```diff
@@ -0,0 +1,120 @@
+const fs = require('fs')
+const path = require('path')
+const globby = require('globby')
+const fastmatter = require('fastmatter')
+const marked = require('marked')
+
+const rulesPaths = globby.sync([`./_rules/*.md`])
+const rulesData = rulesPaths.map(rulePath => {
+  const filename = path.parse(rulePath).base
+  const fileContents = fs.readFileSync(rulePath, { encoding: 'utf-8' })
+  const { attributes, body } = fastmatter(fileContents)
+  return [
+    filename, // used for dynamically creating title of testcases
+    { attributes, body }
+  ]
+})
+
+/**
+ * Get headings  from a given body of text
+ * @method getAllHeadingsFromMarkdownBody
+ * @param {String} body text content of markdown file
+ * @returns {Array<Object>} list of headings
+ */
+const getAllHeadingsFromMarkdownBody = (body) => {
+  const lexer = new marked.Lexer({})
+  const tokens = lexer.lex(body)
+  return tokens.reduce((out, token, index) => {
+    const { type } = token
+    if (type === `heading`) {
+      out.push({
+        ...token,
+        number: index // populate with line number, to check for ordering
+      })
+    }
+    return out
+  }, [])
+}
+
+/**
+ * Get `text` of headings of given depth
+ * @param {Array<Object>} headings list of headings
+ * @param {Number} headingDepth given depth of heading
+ * @returns {Array<String>} al headings matching given depth
+ */
+const getHeadingOfDepth = (headings, headingDepth) => {
+  return headings
+    .filter(({ depth }) => depth === headingDepth)
+    .map(({ text }) => text)
+}
+
+test(`should contain rules`, () => {
+  expect(rulesPaths).toBeDefined()
+  expect(rulesPaths.length).toBeGreaterThan(0);
+})
+
+describe(`validate rules frontmatter > attributes`, () => {
```

Comment:
These tests aren't well organised. Each rule should be treated as its own thing, rather than as a collection. You also should isolate each of the tests, rather than putting a whole bunch of expects into a single test. That's fine if they all amount to the same thing, but these are very much separate things you are testing. And you need to avoid putting all of your tests in a single file.

Here's how I suggest to do it:

```js
// utils/describe-rule.js
function describeRule (groupName, runTests) {
  rules.forEach(ruleData => {
    // All describeRule calls are grouped by rule at the top, and groupName second
    describe(ruleData.fileName, () => {
      describe(groupName, () => {
        runTests(ruleData)
      })
    })
  }
}

// rule-frontmatter.test.js
describeRule('frontmatter', ({ frontmatter }) => {=
  const requiredProps = ['id', 'name', 'etc']

  test.each(requiredProps)('has required property `%s`', (requiredProp) => {
    expect(frontmatter).toHaveProperty(requiredProp);
  })
  /* ... */
})

// rule-headings.test.js
describeRule('headings', ({ body }) => {
  const headings = getAllHeadingsFromMarkdownBody(body)
  const level2Headings = getHeadingOfDepth(headings, 2)
  const level3Headings = getHeadingOfDepth(headings, 3)

  const requiredH2s = [
    `Applicability`,
    `Assumptions`,
    `Accessibility Support`,
    `Background`,
    `Test Cases`
  ]
  const requiredH3s = [`Passed`, `Failed`, `Inapplicable`]
  
  test.each(requiredH2s)('has required H2 `%s`', (requiredH2) => {
    expect(level2Headings).toContain(requiredH2);
  })
  
  test.each(requiredH3s)('has required H3 `%s`', (requiredH3) => {
    expect(level3Headings).toContain(requiredH3);
  })

  test('are in the expected order')
})
```